src/shared/regions/endpointsProvider.ts

```diff
@@ -0,0 +1,53 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import { getLogger, Logger } from '../logger'
+import { ResourceFetcher } from '../resourcefetcher/resourcefetcher'
+import { Endpoints, loadEndpoints } from './endpoints'
+
+export class EndpointsProvider {
+    private readonly logger: Logger = getLogger()
+    public get onEndpointsUpdated(): vscode.Event<EndpointsProvider> {
+        return this.onEndpointsUpdatedEmitter.event
+    }
+    private readonly onEndpointsUpdatedEmitter: vscode.EventEmitter<EndpointsProvider> = new vscode.EventEmitter()
+    private endpoints: Endpoints | undefined
+
+    /**
+     * @param localFetcher Retrieves endpoints manifest from local sources available to the toolkit. Expected
+     *                      to resolve fast, and is both a placeholder until the remote resources are loaded, and
+     *                      is a fallback in case the toolkit is unable to load a remote resource
+     * @param remoteFetcher Retrieves endpoints manifest from remote host
+     */
+    public constructor(
+        private readonly localFetcher: ResourceFetcher,
+        private readonly remoteFetcher: ResourceFetcher
+    ) {}
+
+    public getEndpoints(): Endpoints | undefined {
```

Comment:
Could we have empty endpoints instead of undefined? is undefined useful here?

Reply:
Hard to say. Right now it implies "not loaded yet". If more fields are added to the Endpoints structure, it might not be possible to have an "empty" one. Calling code might be simpler for not handling undefined though...